### ISSUE TITLE
message handler error no json object could be decoded 

### ISSUE THREAD

**Issue by ahmed-alhaj1** [miguelgrinberg/python-socketio#199]
* Restarting with stat
 * Debugger is active!
 * Debugger PIN: 304-783-885
(5231) wsgi starting up on http://127.0.0.1:5000
(5231) accepted ('127.0.0.1', 38358)
127.0.0.1 - - [06/Aug/2018 11:16:00] "GET /socket.io/?EIO=3&transport=polling&t=1533568560130-24 HTTP/1.1" 200 381 0.000431
message handler error
Traceback (most recent call last):
  File "/home/alhaj1/fenv/local/lib/python2.7/site-packages/engineio/server.py", line 411, in _trigger_event
    return self.handlers[event](*args)
  File "/home/alhaj1/fenv/local/lib/python2.7/site-packages/socketio/server.py", line 516, in _handle_eio_message
    pkt = packet.Packet(encoded_packet=data)
  File "/home/alhaj1/fenv/local/lib/python2.7/site-packages/socketio/packet.py", line 43, in __init__
    self.attachment_count = self.decode(encoded_packet)
  File "/home/alhaj1/fenv/local/lib/python2.7/site-packages/socketio/packet.py", line 113, in decode
    self.data = self.json.loads(ep)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
127.0.0.1 - - [06/Aug/2018 11:16:00] "POST /socket.io/?EIO=3&transport=polling&t=1533568560137-25&sid=1242e199bae14c41944ce64d930b2205 HTTP/1.1" 200 219 0.003187
(5231) accepted ('127.0.0.1', 38360)
(5231) accepted ('127.0.0.1', 38362)
127.0.0.1 - - [06/Aug/2018 11:16:00] "GET /socket.io/?EIO=3&transport=polling&t=1533568560138-26&sid=1242e199bae14c41944ce64d930b2205 HTTP/1.1" 200 215 0.000627
                
server side 

import os
from flask import Flask, render_template, request, send_from_directory
from flask_socketio import SocketIO
import json as js

app = Flask(__name__)
#json_str = json_str.encode('latin-1').decode('utf-8')
socketio = SocketIO(app, async_mode='eventlet')
basedir = os.path.abspath(os.path.dirname(__file__))

app.config['SECURITY_KEY'] = 'secret'
app.config['DEBUG'] = True
app.config['ALLOWED_EXTENSIONS'] = set(["pdf", "docx", "doc"])

@app.route('/')
def show_file():
        return render_template("display.html")

@socketio.on_error()        # Handles the default namespace
def error_handler(e):
    pass

@socketio.on('json')
def handle_message(json):

    
    print( 'recieved data', str(json))

if __name__ == '__main__':
	socketio.run(app)
client 

<script type="text/javascript" src="//code.jquery.com/jquery-2.1.3.min.js"></script>

<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/socket.io/1.3.6/socket.io.min.js"></script>
<script type="text/javascript" charset="utf-8">
	
    var  namespace = "/"
    var data = "name its connected";
    //var data= "Hellow world woooooooooooooow";
    var socket = io.connect('http://' + location.host + ':' + location.port );
    socket.on('connect', function() {
        socket.emit('client_connected',{'data':'you are connected!'});
    });
</script>

</body>
</html>

**Comment by miguelgrinberg** [comment]
It's very hard to follow your code when it is all badly formatted. Do you have a complete example I can install and run here?

**Comment by ahmed-alhaj1** [comment]
server.py

from flask_socketio import SocketIO, send, emit
import json

app = Flask(__name__)
socketio = SocketIO(app)
basedir = os.path.abspath(os.path.dirname(__file__))

app.config['SECURITY_KEY'] = 'secret'
app.config['DEBUG'] = True
app.config['ALLOWED_EXTENSIONS'] = set(["pdf", "docx", "doc"])


@app.route('/')
def show_file():
        return render_template("display.html")

@socketio.on('client_connected')
def handle_message(json):

    print( 'connection started ', str(json))

if __name__ == '__main__':
##################################################################################
##################################################################################
client.html

<!DOCTYPE html>
<html>
<head> <h1> convert get pdf contents </h1>

</head>
<body>

<script type="text/javascript" src="//code.jquery.com/jquery-2.1.3.min.js"></script>

<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/socket.io/1.3.6/socket.io.min.js"></script>


<title> "test socket io "</title>

<script>
    var socket = io.connect('http://' + location.host + ':' + location.port );
    socket.on('connect', function() {
        socket.emit('client_connected',{data:'you are connected!'});
    });
</script>

</body>
</html>

**Comment by ahmed-alhaj1** [comment]
client.html

`

<!DOCTYPE html>
<html>

<body>

<script type="text/javascript" src="//code.jquery.com/jquery-2.1.3.min.js"></script>

<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/socket.io/1.3.6/socket.io.min.js"></script>


<title> "test socket io "</title>

<script>
    var socket = io.connect('http://' + location.host + ':' + location.port );
    socket.on('connect', function() {
        socket.emit('client_connected',{data:'you are connected!'});
    });
</script>

</body>
</html>      
`

**Comment by ahmed-alhaj1** [comment]
sorry I was able to past html the html code dispeaer
https://shrib.com/#dsiplay.html

**Comment by miguelgrinberg** [comment]
You are using a very old version of the Socket.IO JS client. Does this problem occur with the latest version?

**Comment by ahmed-alhaj1** [comment]
I tried this is still the same error ValueError: No JSON object could be decoded
https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.1.1/socket.io.dev.js

**Comment by miguelgrinberg** [comment]
To connect, use the following:

    var socket = io.connect('http://' + location.host + ':' + location.port );

The way you are doing it makes the port appear twice in the URL.

**Comment by ahmed-alhaj1** [comment]
Thank so much it worked
